### PR TITLE
No supershell for Emacs and other color supporting dumb terminals

### PR DESCRIPTION
**Pull request by hvesalai**
As the title says

### CODE REVIEW

File: main/src/main/scala/sbt/internal/SysProp.scala [hunk 1]
```diff
@@ -89,7 +89,8 @@ object SysProp {
 
   def fileCacheSize: Long =
     SizeParser(System.getProperty("sbt.file.cache.size", "128M")).getOrElse(128L * 1024 * 1024)
-  def supershell: Boolean = booleanOpt("sbt.supershell").getOrElse(color)
+  def insideEmacs: Boolean = sys.env.contains("INSIDE_EMACS")
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/internal/SysProp.scala]: I am not against checking specifically for Emacs here, but I wonder if it makes sense here to check for `TERM` set to `"dumb"`, which supposedly is the value set on Emacs shell according to @SethTisue.

Comment (hvesalai) [reply main/src/main/scala/sbt/internal/SysProp.scala]: Yes, that's the other option. I can change the PR to do that if we so choose

```
scala> sys.env.get("TERM")
res0: Option[String] = Some(dumb)
```

Comment (eed3si9n) [reply main/src/main/scala/sbt/internal/SysProp.scala]: I think checking for `TERM` is preferable since the notion could apply to other environments.

**Review by eed3si9n — APPROVED** [review]
Thanks @hvesalai!

**Pull request merged**